go/vt/vtgate/vindexes/null.go

```diff
@@ -0,0 +1,72 @@
+/*
+Copyright 2017 Google Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package vindexes
+
+import (
+	"bytes"
+
+	"github.com/youtube/vitess/go/sqltypes"
+)
+
+var (
+	_        Functional = (*Null)(nil)
+	nullksid            = []byte{0}
+)
+
+// Null defines a vindex that always return 0. It's Unique and
+// Functional.
+// This is useful for rows that always go into the first shard.
```

Comment:
It will be good to explain the use case here. Something like:
"This Vindex can be used for validating an unsharded->sharded transition. Unlike other vindexes, this one will work even for NULL input values. This will allow you to keep MySQL auto-inc columns unchanged."